crates/biome_formatter/src/format_element.rs

```diff
@@ -341,6 +349,15 @@ pub trait FormatElements {
     /// lines if this element is part of a group and the group doesn't fit on a single line.
     fn will_break(&self) -> bool;
 
+    /// Returns true if this [FormatElement] has the potential to break across multiple lines when printed.
+    /// This is the case _only_ if this format element recursively contains a [FormatElement::Line].
+    ///
+    /// It's possible for [FormatElements::will_break] to return true while this function returns false,
+    /// such as when the group contains a [crate::builders::expand_parent] or some text within the group
+    /// contains a newline. Neither of those cases directly contain a [FormatElement::Line], and so they
+    /// do not _directly_ break.
+    fn may_directly_break(&self) -> bool;
```

Comment:
Honestly I don't know which is the better name between `will_break`, `will_directly_break` or `will_indirectly_break`...because they're all both accurate _and_ inaccurate in different ways lol. `indirectly` i think implies that the element won't break itself, but some child element does, or something like that. `directly` implies that the element itself does, but children might also break instead. the plain `will_break` at least covers both of those, despite being a little more ambiguous to read.